### PR TITLE
Use ISymUnmanaged* from Microsoft.DiaSymReader

### PR DESCRIPTION
**Pull request by amcasey**
Rebased from https://github.com/tmat/roslyn/tree/DiaSymReader and updated for new sources.

### CODE REVIEW

**Comment by amcasey** [comment]
FYI @cston @KevinH-MS @ManishJayaswal

File: src/Test/PdbUtilities/Shared/SymUnmanagedSequencePoint.cs [hunk 1]
```diff
@@ -2,48 +2,10 @@
 
 using System;
 using System.Collections.Immutable;
-using System.Diagnostics;
-using System.Runtime.InteropServices;
 using Microsoft.CodeAnalysis;
 
-namespace Microsoft.VisualStudio.SymReaderInterop
+namespace Microsoft.DiaSymReader
```
Comment (cston) [inline src/Test/PdbUtilities/Shared/SymUnmanagedSequencePoint.cs]: Should this be Roslyn.Test.PdbUtilities instead?

**Comment by cston** [comment]
LGTM

File: src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/CompilationContext.cs [hunk 2]
```diff
@@ -7,7 +7,7 @@
 using Microsoft.CodeAnalysis.Emit;
 using Microsoft.CodeAnalysis.ExpressionEvaluator;
 using Microsoft.VisualStudio.Debugger.Evaluation.ClrCompilation;
-using Microsoft.VisualStudio.SymReaderInterop;
+using Microsoft.DiaSymReader;
```
Comment (VladimirReshetnikov) [inline src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/CompilationContext.cs]: Consider sorting usings.

File: src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/EvaluationContext.cs [hunk 3]
```diff
@@ -15,7 +15,7 @@
 using Microsoft.CodeAnalysis.Emit;
 using Microsoft.CodeAnalysis.ExpressionEvaluator;
 using Microsoft.VisualStudio.Debugger.Evaluation;
-using Microsoft.VisualStudio.SymReaderInterop;
+using Microsoft.DiaSymReader;
```
Comment (VladimirReshetnikov) [inline src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/EvaluationContext.cs]: Consider sorting usings.

File: src/ExpressionEvaluator/CSharp/Test/ExpressionCompiler/ExpressionCompilerTestBase.cs [hunk 4]
```diff
@@ -15,7 +15,7 @@
 using Microsoft.VisualStudio.Debugger.Evaluation;
 using Microsoft.VisualStudio.Debugger.Evaluation.ClrCompilation;
 using Roslyn.Test.Utilities;
-using Microsoft.VisualStudio.SymReaderInterop;
+using Microsoft.DiaSymReader;
```
Comment (VladimirReshetnikov) [inline src/ExpressionEvaluator/CSharp/Test/ExpressionCompiler/ExpressionCompilerTestBase.cs]: Consider sorting usings.

File: src/ExpressionEvaluator/CSharp/Test/ExpressionCompiler/UsingDebugInfoTests.cs [hunk 5]
```diff
@@ -11,7 +11,7 @@
 using Microsoft.CodeAnalysis.CSharp.Symbols;
 using Microsoft.CodeAnalysis.ExpressionEvaluator;
 using Microsoft.VisualStudio.Debugger.Evaluation.ClrCompilation;
-using Microsoft.VisualStudio.SymReaderInterop;
+using Microsoft.DiaSymReader;
```
Comment (VladimirReshetnikov) [inline src/ExpressionEvaluator/CSharp/Test/ExpressionCompiler/UsingDebugInfoTests.cs]: Consider sorting usings.

File: src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/CSharpExpressionCompiler.csproj [hunk 6]
```diff
@@ -19,10 +19,12 @@
   </PropertyGroup>
```
Comment (VladimirReshetnikov) [inline src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/CSharpExpressionCompiler.csproj]: Consider adding ` <!-- Copyright (c)  Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information. -->` at the top of the file.

File: src/ExpressionEvaluator/CSharp/Test/ExpressionCompiler/CSharpExpressionCompilerTest.csproj [hunk 7]
```diff
@@ -76,14 +76,15 @@
   </PropertyGroup>
```
Comment (VladimirReshetnikov) [inline src/ExpressionEvaluator/CSharp/Test/ExpressionCompiler/CSharpExpressionCompilerTest.csproj]: Consider adding ` <!-- Copyright (c)  Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information. -->` at the top of the file.

File: src/ExpressionEvaluator/Core/Source/ExpressionCompiler/ExpressionCompiler.csproj [hunk 8]
```diff
@@ -24,12 +24,6 @@
     <Compile Include="..\..\..\..\Test\PdbUtilities\Shared\CustomDebugInfoReader.cs">
```
Comment (VladimirReshetnikov) [inline src/ExpressionEvaluator/Core/Source/ExpressionCompiler/ExpressionCompiler.csproj]: Consider adding ` <!-- Copyright (c)  Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information. -->` at the top of the file.

File: src/ExpressionEvaluator/Core/Test/ExpressionCompiler/ExpressionCompilerTestUtilities.csproj [hunk 9]
```diff
@@ -22,10 +22,8 @@
     <Reference Include="System" />
```
Comment (VladimirReshetnikov) [inline src/ExpressionEvaluator/Core/Test/ExpressionCompiler/ExpressionCompilerTestUtilities.csproj]: Consider adding ` <!-- Copyright (c)  Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information. -->` at the top of the file.

File: src/ExpressionEvaluator/Core/Test/ExpressionCompiler/MethodDebugInfo.cs [hunk 10]
```diff
@@ -3,7 +3,7 @@
 using System.Collections.Immutable;
 using System.Linq;
 using Roslyn.Utilities;
-using Microsoft.VisualStudio.SymReaderInterop;
+using Microsoft.DiaSymReader;
```
Comment (VladimirReshetnikov) [inline src/ExpressionEvaluator/Core/Test/ExpressionCompiler/MethodDebugInfo.cs]: Consider sorting usings.

File: src/ExpressionEvaluator/VisualBasic/Source/ExpressionCompiler/BasicExpressionCompiler.vbproj [hunk 11]
```diff
@@ -18,6 +18,7 @@
   <ItemGroup Label="File References">
```
Comment (VladimirReshetnikov) [inline src/ExpressionEvaluator/VisualBasic/Source/ExpressionCompiler/BasicExpressionCompiler.vbproj]: Consider adding ` <!-- Copyright (c)  Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information. -->` at the top of the file.

File: src/ExpressionEvaluator/VisualBasic/Source/ExpressionCompiler/CompilationContext.vb [hunk 12]
```diff
@@ -11,7 +11,7 @@ Imports Microsoft.CodeAnalysis.VisualBasic.Symbols
 Imports Microsoft.CodeAnalysis.VisualBasic.Symbols.Metadata.PE
 Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
 Imports Microsoft.VisualStudio.Debugger.Evaluation.ClrCompilation
-Imports Microsoft.VisualStudio.SymReaderInterop
+Imports Microsoft.DiaSymReader
```
Comment (VladimirReshetnikov) [inline src/ExpressionEvaluator/VisualBasic/Source/ExpressionCompiler/CompilationContext.vb]: Consider sorting Imports.

File: src/ExpressionEvaluator/VisualBasic/Test/ExpressionCompiler/BasicExpressionCompilerTest.vbproj [hunk 13]
```diff
@@ -19,6 +19,7 @@
   <ItemGroup Label="File References">
```
Comment (VladimirReshetnikov) [inline src/ExpressionEvaluator/VisualBasic/Test/ExpressionCompiler/BasicExpressionCompilerTest.vbproj]: Consider adding ` <!-- Copyright (c)  Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information. -->` at the top of the file.

File: src/ExpressionEvaluator/VisualBasic/Test/ExpressionCompiler/ExpressionCompilerTests.vb [hunk 14]
```diff
@@ -11,7 +11,7 @@ Imports Microsoft.CodeAnalysis.VisualBasic.ExpressionEvaluator
 Imports Microsoft.CodeAnalysis.VisualBasic.Symbols
 Imports Microsoft.VisualStudio.Debugger.Evaluation
 Imports Microsoft.VisualStudio.Debugger.Evaluation.ClrCompilation
-Imports Microsoft.VisualStudio.SymReaderInterop
+Imports Microsoft.DiaSymReader
```
Comment (VladimirReshetnikov) [inline src/ExpressionEvaluator/VisualBasic/Test/ExpressionCompiler/ExpressionCompilerTests.vb]: Consider sorting Imports.

File: src/VisualStudio/Core/Def/ServicesVisualStudio.csproj [hunk 15]
```diff
@@ -81,36 +81,6 @@
     <Compile Include="..\..\..\Test\PdbUtilities\Shared\IMetadataImport.cs">
```
Comment (VladimirReshetnikov) [inline src/VisualStudio/Core/Def/ServicesVisualStudio.csproj]: Consider adding ` <!-- Copyright (c)  Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information. -->` at the top of the file.

**Comment by tmat** [comment]
:+1:

File: src/Test/PdbUtilities/Shared/SymUnmanagedSequencePoint.cs (hunk 1 above)
Comment (amcasey) [inline src/Test/PdbUtilities/Shared/SymUnmanagedSequencePoint.cs]: While I agree that it's strange to use a namespace from another assembly, in this particular case, essentially every consumer of the namespace will also want the extension methods.

**Pull request merged**